SC2103 not silenced by `set -e`

Using `set -e` silences [SC2164](https://github.com/koalaman/shellcheck/wiki/SC2164), on the grounds that with `set -e`, the `cd` command cannot fail. However [SC2103](https://github.com/koalaman/shellcheck/wiki/SC2103) is not silenced by `set -e`, despite the rationale being effectively the same.

--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -2133,6 +2133,8 @@ prop_checkCdAndBack2 = verifyNot checkCdAndBack "for f in *; do cd $f || continu
 prop_checkCdAndBack3 = verifyNot checkCdAndBack "while [[ $PWD != / ]]; do cd ..; done"
 prop_checkCdAndBack4 = verify checkCdAndBack "cd $tmp; foo; cd -"
 prop_checkCdAndBack5 = verifyNot checkCdAndBack "cd ..; foo; cd .."
+prop_checkCdAndBack6 = verify checkCdAndBack "for dir in */; do cd \"$dir\"; some_cmd; cd ..; done"
+prop_checkCdAndBack7 = verifyNot checkCdAndBack "set -e; for dir in */; do cd \"$dir\"; some_cmd; cd ..; done"
 checkCdAndBack params = doLists
   where
     shell = shellType params
@@ -2163,14 +2165,13 @@ checkCdAndBack params = doLists
                 else findCdPair (b:rest)
             _ -> Nothing
 
-
     doList list =
-        let cds = filter ((== Just "cd") . getCmd) list in
-            potentially $ do
-                cd <- findCdPair cds
-                return $ info cd 2103 message
-
-    message = "Use a ( subshell ) to avoid having to cd back."
+        if hasSetE params
+        then return ()
+        else let cds = filter ((== Just "cd") . getCmd) list
+             in  potentially $ do
+                     cd <- findCdPair cds
+                     return $ info cd 2103 "Use a ( subshell ) to avoid having to cd back."
 
 prop_checkLoopKeywordScope1 = verify checkLoopKeywordScope "continue 2"
 prop_checkLoopKeywordScope2 = verify checkLoopKeywordScope "for f; do ( break; ); done"

